com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs

```diff
@@ -67,11 +63,24 @@ public StackingSensor(ISensor wrapped, int numStackedObservations)
             }
 
             // TODO support arbitrary stacking dimension
-            m_Shape[0] *= numStackedObservations;
+            m_Shape[m_Shape.Length - 1] *= numStackedObservations;
             m_StackedObservations = new float[numStackedObservations][];
-            for (var i = 0; i < numStackedObservations; i++)
+            m_StackedCompressedObservations = new byte[numStackedObservations][];
+            m_EmptyPNG = CreateEmptyPNG();
```

Comment:
Move inside the `else` block, and check the compression type. I'm surprised this isn't breaking tests, since we'd be accessing the shape out of bounds for a vector sensor.